api/src/main/kotlin/edu/wgu/osmt/elasticsearch/SearchController.kt

```diff
@@ -142,6 +154,52 @@ class SearchController @Autowired constructor(
             .body(searchHits.map { it.content }.toList())
     }
 
+    @Transactional(readOnly = true)
+    @GetMapping(RoutePaths.EXPORT_SKILLS, produces = ["text/csv"])
+    @ResponseBody
+    fun scrollSkills(
+        uriComponentsBuilder: UriComponentsBuilder,
+        @RequestParam(
+            required = false, defaultValue = PublishStatus.DEFAULT_API_PUBLISH_STATUS_SET
+        ) status: Array<String>,
+        @RequestParam(required = false) sort: String?,
+        @AuthenticationPrincipal user: Jwt?
+    ): ResponseEntity<StreamingResponseBody> {
+        if (!appConfig.allowPublicSearching && user === null) {
+            throw GeneralApiException("Unauthorized", HttpStatus.UNAUTHORIZED)
+        }
+
+        val publishStatuses = status.mapNotNull {
+            val status = PublishStatus.forApiValue(it)
+            if (user == null && (status == PublishStatus.Deleted || status == PublishStatus.Draft)) null else status
+        }.toSet()
+        val pageable = PageRequest.of(0, SEARCH_BY_API_THRESHOLD)
+        val responseHeaders = HttpHeaders()
+        responseHeaders.add("Content-Type", "text/csv")
+
+        val countByApiSearch = richSkillEsRepo.countByApiSearch(
+            ApiSearch(""),
```

Comment:
This empty string and the one below are kind of magic. Would you please make a constant or at least a variable for "all RSDs query string" or something like that?

Reply:
sure thing